integration-tests/xml-jaxp/src/main/java/apache/camel/quarkus/component/xml/jaxp/XmlJaxpRoutes.java

```diff
@@ -0,0 +1,146 @@
+import javax.xml.stream.XMLStreamReader;
+import javax.xml.stream.XMLStreamWriter;
+import javax.xml.transform.Source;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.sax.SAXSource;
+import javax.xml.transform.stax.StAXSource;
+import javax.xml.transform.stream.StreamSource;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.util.xml.BytesSource;
+
+public class XmlJaxpRoutes extends RouteBuilder {
+
+    @Override
+    public void configure() throws Exception {
+        from("direct:documentConvert")
+                .convertBodyTo(Document.class)
+                .process(exchange -> {
+                    Document document = exchange.getMessage().getBody(Document.class);
+                    Element element = document.createElement("bar");
+                    element.setTextContent("Bar Text");
+                    document.getDocumentElement().appendChild(element);
+                })
+                .convertBodyTo(String.class);
+
+        from("direct:elementConvert")
+                .convertBodyTo(Document.class)
+                .process(exchange -> {
+                    Document document = exchange.getMessage().getBody(Document.class);
+                    Element element = document.createElement("bar");
+                    element.setTextContent("Bar Text");
+                    document.getDocumentElement().appendChild(element);
+                })
+                .convertBodyTo(Element.class)
+                .convertBodyTo(String.class);
+
+        from("direct:byteSourceConvert")
+                .convertBodyTo(byte[].class)
+                .convertBodyTo(BytesSource.class)
+                .convertBodyTo(String.class);
+
+        from("direct:sourceConvert")
+                .convertBodyTo(Source.class)
+                .convertBodyTo(String.class);
+
+        from("direct:domSourceConvert")
+                .convertBodyTo(byte[].class)
+                .convertBodyTo(DOMSource.class)
+                .convertBodyTo(String.class);
+
+        from("direct:saxSourceConvert")
+                .convertBodyTo(SAXSource.class)
+                .convertBodyTo(String.class);
+
+        from("direct:staxSourceConvert")
+                .convertBodyTo(StAXSource.class)
+                .convertBodyTo(String.class);
+
+        from("direct:streamSourceConvert")
+                .convertBodyTo(StreamSource.class)
+                .convertBodyTo(String.class);
+
+        from("direct:streamSourceReaderConvert")
+                .convertBodyTo(StreamSource.class)
+                .convertBodyTo(Reader.class)
+                .convertBodyTo(String.class);
+
+        from("direct:xmlStreamReaderCharsetConvert")
+                .convertBodyTo(XMLStreamReader.class, StandardCharsets.ISO_8859_1.name())
```

Comment:
Changing that charset does not seem to break the test. Does the scenario really test with charset ?

Like the incoming body a String ? The expected type is XMLStreamReader, so the chosen converter would be this [one](https://github.com/apache/camel/blob/main/core/camel-xml-jaxp/src/main/java/org/apache/camel/converter/jaxp/StaxConverter.java#L186) ? So it doesn't support charset ?

If right, maybe we would need to find a way to get ISO-8859-1 encoded byte array as input.

Reply:
I pretty much stole the original code from Camel. I can double check it.

Reply:
Seems I did too much tidying before the PR and removed the critical part of the route.

The issue should be fixed now.